lib/logstash/codecs/fluent.rb

```diff
@@ -51,6 +53,18 @@ def encode(event)
     @on_event.call(event, MessagePack.pack([tag, epochtime, data.merge(LogStash::Event::TIMESTAMP => event.timestamp.to_iso8601)]))
   end # def encode
 
+  def forwardable_tag(event)
+    tag = event.get("tags") || "log"
```

Comment:
can a tag be empty in fluentd protocol? in other words, do we need to assign a default "log" tag when there is no tag in the event?

Reply:
No, `tag` cannot be empty in fluentd protocol.
We need to assign default value for event which has no tag.